edge/pkg/metamanager/dao/upgradedb/confirm_db.go

```diff
@@ -0,0 +1,143 @@
+	nodeUpgradeReqValue := reflect.ValueOf(nodeUpgradeReq)
+	for i := 0; i < nodeUpgradeReqType.NumField(); i++ {
+		fieldType := nodeUpgradeReqType.Field(i)
+		data, err := queryNodeUpgradeTable("Name", fieldType.Name)
+		if err != nil {
+			return commontypes.NodeUpgradeJobRequest{}, err
+		}
+		nodeUpgradeReqValue.FieldByName(data.Name).Set(reflect.ValueOf(data.Value))
+	}
+	return nodeUpgradeReq, nil
+}
+
+// SaveDeviceTwin save NodeTaskRequestField
+func saveNodeUpgradeConfirm(o orm.Ormer, doc *NodeUpgradeConfirmTable) error {
+	err := o.DoTx(func(ctx context.Context, txOrm orm.TxOrmer) error {
+		// insert data
+		// Using txOrm to execute SQL
+		_, e := txOrm.Insert(doc)
+		// if e != nil the transaction will be rollback
+		// or it will be committed
+		return e
+	})
+	if err != nil {
+		klog.Errorf("Something wrong when insert NodeTaskRequest data: %v", err)
+		return err
+	}
+	klog.V(4).Info("insert NodeTaskRequest data successfully")
+	return nil
+}
+
+// SaveNodeTaskRequest save struct of NodeTaskRequest
+func SaveNodeTaskRequest(o orm.Ormer, nodetaskreq types.NodeTaskRequest) error {
+	var metadata *NodeUpgradeConfirmTable
+	nodeTaskReqType := reflect.TypeOf(nodetaskreq)
+	nodeTaskReqValue := reflect.ValueOf(nodetaskreq)
+	for i := 0; i < nodeTaskReqType.NumField(); i++ {
+		fieldType := nodeTaskReqType.Field(i)
+		metadata = &NodeUpgradeConfirmTable{
+			Name:  fieldType.Name,
+			Value: nodeTaskReqValue.FieldByName(fieldType.Name).String(),
+		}
+		err := saveNodeUpgradeConfirm(o, metadata)
+		if err != nil {
+			return fmt.Errorf("Save NodeTaskRequest to DB error:%v", err)
+		}
+	}
+	return nil
+}
+
+// SaveNodeUpgradeJobRequest save struct of NodeUpgradeJobRequest
+func SaveNodeUpgradeJobRequest(o orm.Ormer, nodeUpgradeJobReq commontypes.NodeUpgradeJobRequest) error {
+	var metadata *NodeUpgradeConfirmTable
+	nodeUpgradeJobReqType := reflect.TypeOf(nodeUpgradeJobReq)
+	nodeUpgradeJobReqValue := reflect.ValueOf(nodeUpgradeJobReq)
+	for i := 0; i < nodeUpgradeJobReqType.NumField(); i++ {
+		fieldType := nodeUpgradeJobReqType.Field(i)
+		metadata = &NodeUpgradeConfirmTable{
+			Name:  fieldType.Name,
+			Value: nodeUpgradeJobReqValue.FieldByName(fieldType.Name).String(),
+		}
+		err := saveNodeUpgradeConfirm(o, metadata)
+		if err != nil {
+			return fmt.Errorf("Save NodeUpgradeJobRequest to DB error:%v", err)
+		}
+	}
+	return nil
+}
+
+// DeleteNodeUpgradeConfirmTable delete NodeUpgradeConfirmTable
+func DeleteNodeUpgradeConfirmTable(o orm.Ormer) error {
+	err := o.DoTx(func(ctx context.Context, txOrm orm.TxOrmer) error {
+		// Delete data
+		// Using txOrm to execute SQL
+		_, e := txOrm.QueryTable(&NodeUpgradeConfirmTable{}).Delete()
+		// if e != nil the transaction will be rollback
+		// or it will be committed
+		return e
+	})
+
+	if err != nil {
+		klog.Errorf("Something wrong when deleting NodeTaskRequest data: %v", err)
+		return err
+	}
+	klog.V(4).Info("Delete NodeTaskRequest data successfully")
+	return nil
```

Comment:
Should be `NodeTaskRequest` or `NodeUpgradeConfirm` task?

Reply:
`NodeUPgradeConfirm`